.github/workflows/tests_wf.yml

```diff
@@ -0,0 +1,76 @@
+name: tests_wf
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        required: true 
+        type: string
+      python-version:
+        required: true 
+        type: string
+      test-extras:
+        required: false
+        type: boolean
+        default: false
+      upload-coverage:
+        required: false
+        type: boolean
+        default: false
+  
+env:
+  PYTEST_ARGS: ""
+
+jobs:
+  tests:
+    runs-on: ${{ inputs.os }}-latest
+    steps:
+    - name: Set env variables
+      if: inputs.upload-coverage
+      run: echo "PYTEST_ARGS=--cov-append --cov=pydicom" >> "$GITHUB_ENV" 
+    - name: Check-out code
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 2
+    - name: Set up Python ${{ inputs.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ inputs.python-version }}
+        cache: pip
+        allow-prereleases: true
+
+    - name: Install pydicom and tools  # not 'editable install'
+      run: |
+        python -m pip install .[dev]
+
+    - name: Test without numpy or handlers, incl pydicom-data (in [dev])
+      run: |
+        python -m pytest --cov-reset ${{ env.PYTEST_ARGS }}
+
+    - name: Test with Numpy
+      run: |
+        python -m pip install .[basic]
+        python -m pytest ${{ env.PYTEST_ARGS }} 
+
+    - name: Test with all optional dependencies
+      if: inputs.test-extras
+      run: |
+        python -m pip install .[pixeldata,gpl-license]
+        python -m pytest ${{ env.PYTEST_ARGS }} tests/test_pylibjpeg.py
+        python -m pytest ${{ env.PYTEST_ARGS }} tests/test_pillow_pixel_data.py
+        python -m pytest ${{ env.PYTEST_ARGS }} tests/test_JPEG_LS_transfer_syntax.py tests/test_jpeg_ls_pixel_data.py
+        python -m pytest ${{ env.PYTEST_ARGS }} tests/test_gdcm_pixel_data.py tests/test_encoders_gdcm.py
+        python -m pip uninstall -y pydicom-data
+
+    - name: Test with missing libs
+      if: inputs.test-extras
+      # A small number of tests only run if libs missing
+      run: |
+        python -m pip uninstall -y pydicom_data pylibjpeg-openjpeg pylibjpeg-libjpeg pylibjpeg-rle
+        python -m pytest ${{ env.PYTEST_ARGS }} tests/test_pylibjpeg.py tests/test_data_manager.py
+
+    - name: Send coverage results
+      if: success() && inputs.upload-coverage
+      uses: codecov/codecov-action@v3
+      with:
+        env_vars: inputs.os, inputs.python-version
```

Comment:
This is nice! I wasn't aware of shared workflows before.

Reply:
Me neither - I think they are relatively new (~a little over a year maybe?).  I find it helps to avoid cluttering the 'main' workflow with all the details, and as mentioned before avoids the duplication between PR and merge workflows, making the matrix choices much easier to see and adapt.